packages/vue/src/create-element.js

```diff
@@ -1,61 +1,71 @@
+import h from 'hastscript'
+import toH from 'hast-to-hyperscript'
+
 /**
  * MDX default components
  */
-const DEFAULTS = {
+const defaults = {
   inlineCode: 'code',
-  wrapper: 'div'
+  wrapper: {
+    name: 'MDXWrapper',
+    render: function (h) {
+      const children = this.$slots.default
+      return children.length === 1 ? children : h('div', {}, children)
+    }
+  }
 }
 
-/**
- * Renders final tag/component
- * @param {Vue.Component|String} type Element or tag to render
- * @param {Object|Array} props Props and attributes for element
- * @param {Array} children Array of child nodes for component
- * @returns {Vue.VNode} VNode of final rendered element
- */
-export default function(type, props, children) {
-
-  const h = this.createElement
-  const components = this.components
-  const defaults = Object.keys(DEFAULTS)
-
-  let tag
-  let elProps = props
-
-  // We check context to see if the element/tag
-  // is provided in the MDXProvider context.
-  if (Object.keys(components).includes(type)) {
-    // We check to see if props is of type object.
-    // If it is, then we pass them into the MDXContext component
-    const componentProps = typeof props === 'object' ? props : undefined
-    tag = components[type](componentProps)
-
-  } else if (defaults.includes(type)) {
-
-    tag = DEFAULTS[type]
-    // Remove components object from attrs
-    const { components, ...attrs } = elProps.attrs
-    elProps = {
-      attrs
-    }
+const own = {}.hasOwnProperty
+
+export default function (type, props, ...children) {
+  let node
+
+  if (own.call(this.components, type)) {
+    type = this.components[type]
+  } else if (own.call(defaults, type)) {
+    type = defaults[type]
+  }
 
-  // Render final tag if component is not provided in context
+  if (props && typeof props === 'object' && !Array.isArray(props)) {
+    // Empty.
   } else {
-    tag = type
-
-    if (['a', 'input', 'img'].includes(tag)) {
-      const { attrs, ...domProps } = elProps
-      const data = {
-        attrs: attrs,
-        domProps
-      }
-
-      elProps = {
-        ...elProps,
-        ...data
-      }
-    }
+    children.unshift(props)
+    props = {}
+  }
+
+  children = children
+    .flatMap(d => (d == null ? [] : d))
+    .map(d =>
+      typeof d === 'number' || typeof d === 'string'
+        ? this.createElement('d', {}, String(d)).children[0]
```

Comment:
this seems a bit quirky.
Why does it create an unused element and pick the first child of that element, rather than creating the element directly?
Would
```suggestion
        ? this.createElement(d)
```

work?

Reply:
It is a bit quirky! The reason is to get a VNode for the text, not for an element

I assumed the above would work the best. It seems there is private function `this._v(d)` we could also use, but I’d rather stay away from hidden stuff.

The reason that we must make them into nodes, is because we’re using `toH` here to handle difference in JSX attribute casing and the complexity of where Vue expects props: https://vuejs.org/v2/guide/render-function.html#The-Data-Object-In-Depth.
But, `toH` deals with hast children but here they are already compiled to vue nodes except for the texts